pyartifactory/objects.py

```diff
@@ -784,19 +812,20 @@ def deploy(
         :param artifact_path: Path to file in Artifactory
         :param local_file_location: Location of the file to deploy
         """
-        artifact_path = artifact_path.lstrip("/")
-        local_filename = artifact_path.split("/")[-1]
-        with open(local_file_location, "rb") as file:
-            self._put(f"{artifact_path}", data=file)
-            logger.debug("Artifact %s successfully deployed", local_filename)
-            return self.info(artifact_path)
-
-    def download(self, artifact_path: str, local_directory_path: str = None) -> str:
-        """
-        :param artifact_path: Path to file in Artifactory
-        :param local_directory_path: Local path to where the artifact will be imported
-        :return: File name
-        """
+        if isdir(local_file_location):
+            for root, _, files in os.walk(local_file_location):
+                new_root = f"{artifact_path}/{root[len(local_file_location):]}"
+                for file in files:
+                    self.deploy(join(root, file), f"{new_root}/{file}")
+        else:
+            artifact_path = artifact_path.lstrip("/")
+            local_filename = artifact_path.split("/")[-1]
+            with open(local_file_location, "rb") as file:  # type: ignore[assignment]
+                self._put(f"{artifact_path}", data=file)
+                logger.debug("Artifact %s successfully deployed", local_filename)
+        return self.info(artifact_path)
+
+    def _download(self, artifact_path: str, local_directory_path: str = None) -> str:
```

Comment:
Can you add a docstring for this method. Indeed, according to `PEP 8`:
```
Docstrings are not necessary for non-public methods, but you should have a comment that describes what the method does. This comment should appear after the "def" line.
```

Reply:
I have cloned the docstring ffrom `download` and adjusted both slightly.